### PR TITLE
Adding "Eagle Viper Rep", open sourced Eagle/Viper V2/V3 PCB

### PR DESCRIPTION
**Pull request by 4pplet**
## Description

Adding support for a open sourced alternative PCB for the Viper/Eagle V2/V3 by Duck: [https://github.com/4pplet/eagle_viper_rep](https://github.com/4pplet/eagle_viper_rep)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/4pplet/eagle_viper_rep/rev_a/readme.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Viper/Eagle REP Rev A
+
+Replacement PCB for Viper/Eagle V2/V3
+
+* Keyboard Maintainer: [4pplet](https://github.com/4pplet)
+* Hardware Supported: Viper/Eagle REP Rev A
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/4pplet/eagle_viper_rep/rev_a/readme.md:8]: Could you add instructions on how to enter the bootloader to flash new firmware?

File: keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            setPinOutput(NUM_PIN);
+            writePin(NUM_PIN, 0);
+        }
+        else
+            setPinInputHigh(NUM_PIN);
+    }
+    return res;
+}
+
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c:52]:
```suggestion
layer_state_t layer_state_set_kb(layer_state_t state) {
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    if(DISPLAY_LAYERS){
```
Comment (drashna) [inline keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c:53]:
```suggestion
    state = layer_state_set_user(state);
    if(DISPLAY_LAYERS){
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        setLayerLed(state);
+    }
+    return state;
+}
+/* Set indicator leds to indicate which layer is active */
+void setLayerLed(uint32_t state){
```
Comment (drashna) [inline keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c:59]:
```suggestion
void setLayerLed(layer_state_t state){
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    switch(state){
```
Comment (drashna) [inline keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c:60]:
```suggestion
    switch(get_highest_layer(state)){
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.h [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "quantum.h"
+
+void setLayerLed(uint32_t state);
```
Comment (drashna) [inline keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.h:21]:
```suggestion
void setLayerLed(layer_state_t state);
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/4pplet/eagle_viper_rep/rev_a/readme.md (hunk 1 above)
Comment (4pplet) [reply keyboards/4pplet/eagle_viper_rep/rev_a/readme.md:8]: Instruction on how to enter DFU added 👍

**Review by drashna — APPROVED** [review]

File: keyboards/4pplet/eagle_viper_rep/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+// main layer
+[0] = LAYOUT_all(
+KC_ESC,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,       KC_6,      KC_7,      KC_8,      KC_9,       KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_GRV,
```
Comment (tzarc) [inline keyboards/4pplet/eagle_viper_rep/keymaps/default/keymap.c:22]:
```suggestion
  KC_ESC,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,       KC_6,      KC_7,      KC_8,      KC_9,       KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_GRV,
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/config.h [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4444
```
Comment (Erovia) [inline keyboards/4pplet/eagle_viper_rep/rev_a/config.h:20]: This VID is used by `40percentclub` and `4pplet` and both of them has boards in the VIA repo.
Can you come up with a unique one?

**Review by Erovia — CHANGES_REQUESTED** [review]

File: keyboards/4pplet/eagle_viper_rep/rev_a/chconf.h [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/cannonkeys/db60/chconf.h -r platforms/chibios/common/configs/chconf.h`
+ */
+
```
Comment (tzarc) [inline keyboards/4pplet/eagle_viper_rep/rev_a/chconf.h:21]:
```suggestion
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/chconf.h [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+#pragma once
+
+#define CH_CFG_ST_FREQUENCY 10000
+
+#define CH_CFG_ST_RESOLUTION 16
+
+#define CH_CFG_OPTIMIZE_SPEED FALSE
+
+#define CH_CFG_USE_CONDVARS_TIMEOUT FALSE
```
Comment (tzarc) [inline keyboards/4pplet/eagle_viper_rep/rev_a/chconf.h:30]:
```suggestion
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/halconf.h [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/cannonkeys/db60/halconf.h -r platforms/chibios/common/configs/halconf.h`
+ */
+
```
Comment (tzarc) [inline keyboards/4pplet/eagle_viper_rep/rev_a/halconf.h:21]:
```suggestion
```

**Review by Erovia — APPROVED** [review]

File: keyboards/4pplet/eagle_viper_rep/rev_a/config.h [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+#define BACKLIGHT_PWM_CHANNEL   4
+#define BACKLIGHT_PAL_MODE      2
+#define BACKLIGHT_ON_STATE      1
+#define BACKLIGHT_LEVELS        10
+#define BACKLIGHT_BREATHING     TRUE
+
+/* Underglow */
+#define RGB_DI_PIN  A7
+#define WS2812_SPI  SPID1
+#define WS2812_SPI_MOSI_PAL_MODE 0
```
Comment (tzarc) [inline keyboards/4pplet/eagle_viper_rep/rev_a/config.h:45]:
```suggestion
#define WS2812_SPI_MOSI_PAL_MODE 0
#define WS2812_SPI_SCK_PIN A5
#define WS2812_SPI_SCK_PAL_MODE 0
```

File: keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "rev_a.h"
+
+void board_init(void) {
+    palSetLineMode(A5, PAL_MODE_ALTERNATE(0));
```
Comment (tzarc) [inline keyboards/4pplet/eagle_viper_rep/rev_a/rev_a.c:20]:
```suggestion
```

**Review by tzarc — CHANGES_REQUESTED** [review]

File: keyboards/4pplet/eagle_viper_rep/rev_a/config.h (hunk 8 above)
Comment (4pplet) [reply keyboards/4pplet/eagle_viper_rep/rev_a/config.h:20]: Thanks for letting me know tho, I was not aware that the same VID was used by them.

**Review by tzarc — APPROVED** [review]

**Pull request merged**